src/sage/ext/fast_callable.pyx

```diff
@@ -465,41 +465,73 @@ def fast_callable(x, domain=None, vars=None,
         etb = ExpressionTreeBuilder(vars=vars, domain=domain)
         et = x._fast_callable_(etb)
 
-    if isinstance(domain, sage.rings.abc.RealField):
-        from sage.ext.interpreters.wrapper_rr import Wrapper_rr as builder
-        str = InstructionStream(sage.ext.interpreters.wrapper_rr.metadata,
-                                len(vars),
-                                domain)
-
-    elif isinstance(domain, sage.rings.abc.ComplexField):
-        from sage.ext.interpreters.wrapper_cc import Wrapper_cc as builder
-        str = InstructionStream(sage.ext.interpreters.wrapper_cc.metadata,
-                                len(vars),
-                                domain)
-
-    elif isinstance(domain, sage.rings.abc.RealDoubleField) or domain is float:
-        from sage.ext.interpreters.wrapper_rdf import Wrapper_rdf as builder
-        str = InstructionStream(sage.ext.interpreters.wrapper_rdf.metadata,
-                                len(vars),
-                                domain)
-    elif isinstance(domain, sage.rings.abc.ComplexDoubleField):
-        from sage.ext.interpreters.wrapper_cdf import Wrapper_cdf as builder
-        str = InstructionStream(sage.ext.interpreters.wrapper_cdf.metadata,
-                                len(vars),
-                                domain)
-    elif domain is None:
-        from sage.ext.interpreters.wrapper_py import Wrapper_py as builder
-        str = InstructionStream(sage.ext.interpreters.wrapper_py.metadata,
-                                len(vars))
-    else:
-        from sage.ext.interpreters.wrapper_el import Wrapper_el as builder
-        str = InstructionStream(sage.ext.interpreters.wrapper_el.metadata,
-                                len(vars),
-                                domain)
+    builder, str = _builder_and_stream(vars=vars, domain=domain)
+
     generate_code(et, str)
     str.instr('return')
     return builder(str.get_current())
 
+
+def _builder_and_stream(vars, domain):
```

Comment:
I can see what's going on from the diff, but if someone comes along a year from now, this docstring isn't really going to explain what the function is doing and why. Can you  expand it a little?

Also, if possible, can you test the import-failure code path?

Reply:
This was just mechanical refactoring of code that was nested a bit too deep. Let me see if I remember what it is doing, after a mere 4 days, not a year...

Reply:
> can you test the import-failure code path?

done in afd3be6